src/frontend/servers/y-provider/src/servers/hocusPocusServer.ts

```diff
@@ -104,13 +40,25 @@ export const hocusPocusServer = Server.configure({
       }
 
       can_edit = document.abilities.update;
-    } catch (error) {
-      console.error('onConnect: backend error', error);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        console.error('onConnect: backend error', error.message);
+      }
+
       return Promise.reject(new Error('Backend error: Unauthorized'));
     }
 
     connection.readOnly = !can_edit;
 
+    /*
+     * Unauthenticated users can be allowed to connect
+     * so we flag only authenticated users
+     */
+    try {
+      const user = await getMe(requestHeaders);
+      requestHeaders['x-user-id'] = user.id;
```

Comment:
Is it possible to use the `context` instead instead of modifying the headers ?

Reply:
Yes, done

Reply:
Thanks